area_detector_handlers/eiger.py

```diff
@@ -0,0 +1,158 @@
+
+        Parameters
+        ----------
+        seq_id : int
+            The sequence id of the data
+
+        frame_num: int or None
+            If not None, return the frame_num'th image from this
+            3D array. Useful for when an event is one image rather
+            than a stack. (Editor's note: It's not clear what this original
+            docstring was supposed to mean. Is it *ever* not None?)
+
+        Returns
+        -------
+            A dask array
+        '''
+        master_path = Path(f'{self._file_prefix}_{seq_id}_master.h5').absolute()
+        try:
+            file = self._files[master_path]
+        except KeyError:
+            file = h5py.File(master_path, 'r')
+            self._files[master_path] = file
+
+        # TODO This should be captured in documents, not extracted here.
+        # This code is retained just in case, but it does not do anything other
+        # than set self._md, which has no effect unless the user obtain direct
+        # access to the handler instance by reaching into the Filler's cache of
+        # them.
+        md = {k: file[v][()] for k, v in self.EIGER_MD_LAYOUT.items()}
+        # the pixel mask from the eiger contains:
+        # 1  -- gap
+        # 2  -- dead
+        # 4  -- under-responsive
+        # 8  -- over-responsive
+        # 16 -- noisy
+        pixel_mask = md['pixel_mask']
+        # pixel_mask[pixel_mask>0] = 1
+        # pixel_mask[pixel_mask==0] = 2
+        # pixel_mask[pixel_mask==1] = 0
+        # pixel_mask[pixel_mask==2] = 1
+        md['binary_mask'] = (pixel_mask == 0)
+        md['framerate'] = 1./md['frame_time']
+        self._md = md
+
+        try:
+            # Eiger firmware v1.3.0 and onwards
+            entry = file['entry']['data']
+        except KeyError:
+            # Older firmwares
+            entry = file['entry']
+
+        # Each 'master' file references multiple 'data' files.
+        # We just need to know many there are, but here we make
+        # a sorted list of their names because it can be handy
+        # when things break and we need to debug.
+        data_files = sorted([key for key in entry.keys() if key.startswith("data")])
+
+        to_concatenate = []
+        for i in range(len(data_files)):
+            dataset = entry[f'data_{1 + (i // self._images_per_file):06d}']
+            da = dask.array.from_array(dataset)
+            to_concatenate.append(da)
+        stack = dask.array.concatenate(to_concatenate)
+        if frame_num is None:
+            return stack
+        else:
+            return stack[frame_num % self.images_per_file]
+
+    def get_file_list(self, datum_kwargs_gen):
+        ''' get the file list.
+
+            Receives a list of datum_kwargs for each datum
+        '''
+        filenames = []
+        for dm_kw in datum_kwargs_gen:
+            seq_id = dm_kw['seq_id']
+            new_filenames = glob(self._base_path + "_" + str(seq_id) + "*")
```

Comment:
f-string can be more readable/shorter here.
```suggestion
            new_filenames = glob(f'{self._base_path}_{seq_id}*')
```